packages/template-typescript-minimal/app/pages/home.tsx

```diff
@@ -105,6 +113,9 @@ const Home = ({value}: Props) => {
                     <div className="divider"></div>
                 </div>
                 <div className="panel">
+                    {/* NOTE: React would complain that this content rendered server-side does not match the one on client-side */}
+                    {pageReq && <p>Path of the page&apos;s request: {pageReq.path}</p>}
+
```

Comment:
Good thing you pointed this out. I think it might be worth while to note this caveat in the docs for the hooks. Something along the lines of "_usePageResponse and usePageRequest should not be used for deterministic rendering of you component, this will result in markup mismatches during your applications hydration stage_"

Reply:
If the interface can discourage it, then we won't have to jot down this caveat. I think with the latest update, having a callback for the interface would discourage using the server context within the content of a component.